### PR TITLE
Add Build & deploy to Demo Org

### PR DESCRIPTION
**Pull request by dineshg13**
**Description:** <Describe what has changed.>

Adds pushing to demo account ECR and deploy to otel-demo cluster

**Link to tracking Issue:** https://datadoghq.atlassian.net/browse/OTEL-539 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: ci/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+FROM  486234852809.dkr.ecr.us-east-1.amazonaws.com/base:xenial
+USER root
+
+RUN clean-apt install make \
+            ca-certificates 
+
+COPY --from=build /src/bin/otelcontribcol_* /otelcontribcol
+USER 1001
+ENTRYPOINT ["/otelcontribcol"]
+EXPOSE 4317 55680 55679
```
Comment (songy23) [inline ci/Dockerfile]: Expose 4318 (for HTTP) as well?

**Review by songy23 — APPROVED** [review]

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# =======================================================================
+
+.build-ci-image: &build-ci-image
+  stage: build
+  tags: ["runner:docker"]
+  image: $CI_IMAGE
+  script:
+    - TAG="$IMAGE_TAG_PREFIX-v$CI_COMMIT_SHORT_SHA"
+    - docker build --file $DOCKERFILE --tag registry.ddbuild.io/ci/opentelemetry-collector-contrib:$TAG --label target=staging .
+    - docker push registry.ddbuild.io/ci/opentelemetry-collector-contrib:$TAG
```
Comment (ancostas) [inline .gitlab-ci.yml]: You can remove the script, and add the variables, yielding a common anchor for both:
```
build-ci-image-main:
    <<: *build-ci-image
    script:
        ...

build-ci-image-demo:
    <<: *build-ci-image
    script:
        ...
```

**Review by ancostas — CHANGES_REQUESTED** [review]
Can you link a ticket in the title with some description/deliverable?

**Review by songy23 — APPROVED** [review]

**Comment by gbbr** [comment]
Can you please set a title and a description to this PR? In the current form it looks like a WIP draft that is not yet ready for sharing.

**Comment by gbbr** [comment]
Closing as stale.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -1,10 +1,18 @@
 variables:
-  CI_IMAGE: registry.ddbuild.io/images/docker:20.10.13
-  RESTORE_CACHE_ATTEMPTS: 2
+  # https://github.com/DataDog/cloudops/tree/master/AWS/build-stable/ci/dd-trace-demo
+  CI_PROJECT_NAME: 'dd-trace-demo'
```
Comment (bfung) [inline .gitlab-ci.yml]: Notes: needs to match https://github.com/DataDog/cloudops/blob/master/AWS/build-stable/ci/opentelemetry-collector-contrib/iam.tf#L3

**Review by songy23 — COMMENTED** [review]
```
license header checking failed:
make[2]: *** [Makefile.Common:131: checklicense] Error 1
./ci/scripts/ci-deploy-demo.sh
./ci/scripts/ci-deploy-demo.sh
```

Please fix

**Review by gbbr — APPROVED** [review]

**Pull request merged**